### PR TITLE
fix(install): respect display options

### PR DESCRIPTION
**Pull request by Alizter**
- [x] changelog
- [x] update tests
- [x] add test
- fixes https://github.com/ocaml/dune/issues/7106
- fixes https://github.com/ocaml/dune/issues/4573

### CODE REVIEW

File: test/blackbox-tests/test-cases/install/display.t [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  >  (public_name foo))
+  > EOF
+
+  $ cat >foo.ml <<EOF
+  > let x = "foo"
+  > EOF
+
+  $ dune build @install
+
+dune install should not output any "Installing" messages:
```
Comment (ejgallego) [inline test/blackbox-tests/test-cases/install/display.t:18]: Maybe test both quiet and not quiet to see clearly the difference.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+Unreleased
+----------
+
+- dune install now respects --display=quiet mode (#7116, fixes #7106, @Alizter)
```
Comment (ejgallego) [inline CHANGES.md:4]: fixes #4573

**Comment by Alizter** [comment]
This will also mean that install we use the default display setting which is `quiet` for most users. In order to see the previous behavior you want `--display=short`. I'm not sure if we want this though. However it would make the command more consistent with the other commands.

**Review by ejgallego — APPROVED** [review]
In my opinion we want install to be silent by default, current behavior was a pain, especially in CI logs.

**Comment by ejgallego** [comment]
Some tests are failing @Alizter , you'll have to rebase and fix the tests to pass `--display=short`

**Comment by ejgallego** [comment]
Test suite still broken

**Comment by Alizter** [comment]
Yes, I haven't fixed it yet.

**Comment by rgrinberg** [comment]
Changed it to draft until it's ready

**Comment by Alizter** [comment]
I still need to fix `dune uninstall` but I don't have time for that atm.

**Comment by Alizter** [comment]
This is now ready.

**Comment by rgrinberg** [comment]
Diff looks a little big

**Comment by Alizter** [comment]
Hmm I messed up the rebase. I have fished something out of my reflog, but I am unsure if the tests will pass so going to check the CI.

**Comment by Alizter** [comment]
Diff is fixed (sorry about that). This is now ready.

**Comment by rgrinberg** [comment]
Merged with some minor changes. Thanks.

**Pull request merged**